tests/wakunode_rest/test_rest_store.nim

```diff
@@ -638,3 +638,147 @@ procSuite "Waku v2 Rest API - Store":
+        fakeWakuMessage(@[byte 03], ts = ts(30, timeOrigin)),
+        fakeWakuMessage(@[byte 04], ts = ts(40, timeOrigin)),
+        fakeWakuMessage(@[byte 05], ts = ts(50, timeOrigin)),
+        fakeWakuMessage(@[byte 06], ts = ts(60, timeOrigin)),
+        fakeWakuMessage(@[byte 07], ts = ts(70, timeOrigin)),
+        fakeWakuMessage(@[byte 08], ts = ts(80, timeOrigin)),
+        fakeWakuMessage(@[byte 09], ts = ts(90, timeOrigin)),
+      ]
+    for msg in msgList:
+      require (waitFor driver.put(DefaultPubsubTopic, msg)).isOk()
+
+    let client = newRestHttpClient(initTAddress(restAddress, restPort))
+
+    let remotePeerInfo = peerSwitch.peerInfo.toRemotePeerInfo()
+    let fullAddr = $remotePeerInfo.addrs[0] & "/p2p/" & $remotePeerInfo.peerId
+
+    var pages = newSeq[seq[WakuMessage]](2)
+
+    # Fields that compose a HistoryCursor object
+    var reqPubsubTopic = DefaultPubsubTopic
+    var reqSenderTime = Timestamp(0)
+    var reqStoreTime = Timestamp(0)
+    var reqDigest = waku_store.MessageDigest()
+
+    for i in 0 ..< 2:
+      let response = await client.getStoreMessagesV1(
+        encodeUrl(fullAddr),
+        encodeUrl(reqPubsubTopic),
+        "", # content topics. Empty ignores the field.
+        "", # start time. Empty ignores the field.
+        "", # end time. Empty ignores the field.
+        encodeUrl($reqSenderTime), # sender time
+        encodeUrl($reqStoreTime), # store time
+        reqDigest.toRestStringMessageDigest(),
+          # base64-encoded digest. Empty ignores the field.
+        "3", # page size. Empty implies default page size.
+        "true", # ascending
+      )
+
+      var wakuMessages = newSeq[WakuMessage](0)
+      for j in 0 ..< response.data.messages.len:
+        wakuMessages.add(response.data.messages[j].toWakuMessage())
+
+      pages[i] = wakuMessages
+
+      # populate the cursor for next page
+      if response.data.cursor.isSome():
+        reqPubsubTopic = response.data.cursor.get().pubsubTopic
+        reqDigest = response.data.cursor.get().digest
+        reqSenderTime = response.data.cursor.get().senderTime
+        reqStoreTime = response.data.cursor.get().storeTime
+
+      check:
+        response.status == 200
+        $response.contentType == $MIMETYPE_JSON
+
+    check:
+      pages[0] == msgList[0 .. 2]
+      pages[1] == msgList[3 .. 5]
+
+    # request last third will lead to rate limit rejection
+    var response = await client.getStoreMessagesV1(
+      encodeUrl(fullAddr),
+      encodeUrl(reqPubsubTopic),
+      "", # content topics. Empty ignores the field.
+      "", # start time. Empty ignores the field.
+      "", # end time. Empty ignores the field.
+      encodeUrl($reqSenderTime), # sender time
+      encodeUrl($reqStoreTime), # store time
+      reqDigest.toRestStringMessageDigest(),
+        # base64-encoded digest. Empty ignores the field.
+      "5", # page size. Empty implies default page size.
+      "true", # ascending
```

Comment:
I think we can simplify the test by leaving the default params. I got confused why using `5` in this case but I think is not needed to be a different page size

Reply:
Agree